### DISCUSSION TITLE
Use regular expressions to determine whether the content contains

### DISCUSSION THREAD

**Discussion by tiantian34** [vectordotdev/vector#13876] [Q&A] [answered]
I want to use if / else if / else regular expression to determine whether it contains content to add fields. I don't know how to write it. After testing, it can't be realized. Is there a code example? Thank you

old code
if [request_uri] =~ "png" {
    
  }
How to write code with remap

**Comment by tobz** [comment, marked as answer]
@tiantian34 There are VRL functions for checking if a string matches a regular expression, but there's also a few other string-related functions for checking for if a string contains a substring, which may be beneficial to you given your example:

- [`contains`](https://vector.dev/docs/reference/vrl/functions/#contains) will allow you to check to see if a substring is contained within a string
- [`starts_with`](https://vector.dev/docs/reference/vrl/functions/#starts_with)/[`ends_with`](https://vector.dev/docs/reference/vrl/functions/#ends_with) will let you check if a string starts with, or ends with, a specific substring
- [`match`](https://vector.dev/docs/reference/vrl/functions/#match) allows you to check if a string matches a regular expression (and [`match_any`](https://vector.dev/docs/reference/vrl/functions/#match_any) lets you check against multiple regular expressions at the same time)

You can see some examples of using the `remap` transform, including writing VRL, in the examples section here: https://vector.dev/docs/reference/configuration/transforms/remap/#examples.
